src/zen/kbs/ZenKeyboardShortcuts.mjs

```diff
@@ -1231,6 +1289,46 @@ var gZenKeyboardShortcutsManager = {
     this.triggerShortcutRebuild();
   },
 
+  _registerPrecedentShortcut(shortcut, browser) {
+    const listener = (event) => {
```

Comment:
A new event is being made on every key added, I would add a single event handler and check on a list of known precedent shortcuts